### PR TITLE
hydrographic_msgs: 1.1.0-1 in 'noetic/distribution.yaml' [bloom]

### PR DESCRIPTION
**Pull request by vatanaksoytezer**
Increasing version of package(s) in repository `hydrographic_msgs` to `1.1.0-1`:

- upstream repository: https://github.com/apl-ocean-engineering/hydrographic_msgs.git
- release repository: https://github.com/apl-ocean-engineering/hydrographic_msgs-release.git
- distro file: `noetic/distribution.yaml`
- bloom version: `0.11.2`
- previous version for package: `null`

## acoustic_msgs

```
* Fix ProjectedSonarImage comments about frame (#26 <https://github.com/apl-ocean-engineering/hydrographic_msgs/issues/26>)
* Add CI, pre-commit, prerelease and dependabot (#27 <https://github.com/apl-ocean-engineering/hydrographic_msgs/issues/27>)
  Co-authored-by: Laura Lindzey <mailto:lindzey@uw.edu>
* Merge pull request #25 <https://github.com/apl-ocean-engineering/hydrographic_msgs/issues/25> from apl-ocean-engineering/add_migration_rule
  Add migration rule from v0.0.1 to 28ec6a7
* Add migration rule from v0.0.1 to 28ec6a7
* Merge pull request #22 <https://github.com/apl-ocean-engineering/hydrographic_msgs/issues/22> from rolker/fix_variable_name
* Rename num_beams to beam_count.
* Merge pull request #21 <https://github.com/apl-ocean-engineering/hydrographic_msgs/issues/21> from CaptKrasno/acoustic_msg_devel
* Contributors: CaptKrasno, Laura Lindzey, Roland Arsenault, Vatan Aksoy Tezer, lauralindzey
```

## environmental_msgs

```
* Add CI, pre-commit, prerelease and dependabot (#27 <https://github.com/apl-ocean-engineering/hydrographic_msgs/issues/27>)
  Co-authored-by: Laura Lindzey <mailto:lindzey@uw.edu>
* Fix typo in package.xml
* Contributors: Laura Lindzey, Vatan Aksoy Tezer
```

### CODE REVIEW

**Comment by cottsay** [comment]
[Holding for Noetic sync](https://discourse.ros.org/t/preparing-for-noetic-sync-2022-11-30/)

**Comment by nuclearsandwich** [comment]
New package review:

- [ ] At least one of the following must be present
  - [ ] Top level license file: 
  - [ ] Per package license files:
- [ ] License is [OSI-approved](https://opensource.org/licenses): BSD is ambiguous especially without a license file in the repository
- [ ] License correctly listed in package.xmls
  - [ ] environmental_msgs: https://github.com/apl-ocean-engineering/hydrographic_msgs/blob/main/environmental_msgs/package.xml#L11
  - [ ] acoustic_msgs: https://github.com/apl-ocean-engineering/hydrographic_msgs/blob/main/acoustic_msgs/package.xml#L11
- [ ] Public source repo: SOURCE_REPO_URL
- [ ] Source repository contains ROS packages and each package meets [REP-144](https://www.ros.org/reps/rep-0144.html) naming conventions
  - [ ] acoustic_msgs
  - [ ] environmental_msgs

**Comment by nuclearsandwich** [comment]
Before this repository can be released some changes need to be made.

@vatanaksoytezer this repository is missing a license file and as a result the package.xml license entry `BSD` is ambiguous. Consider using a SPDX identifier instead of just `BSD`.

I'd like to get a second opinion on the package names here I'm right on the fence about whether "acoustic" and "environmental" are sufficient especially when neither is implicitly underwater. @cottsay what are your thoughts?

**Review by nuclearsandwich — CHANGES_REQUESTED** [review]
License information in the upstream repository is incomplete and these packages have not yet passed naming review.

**Comment by vatanaksoytezer** [comment]
@nuclearsandwich sounds good. I'm just helping for release, tagging @lauralindzey, who is the actual maintainer for opinions on licensing and packaging names.

**Comment by vatanaksoytezer** [comment]
@nuclearsandwich we added the license information in https://github.com/apl-ocean-engineering/hydrographic_msgs/pull/29, do we need to re-release, re-tag or are we good for now?

**Comment by lauralindzey** [comment]
@nuclearsandwich -- what's involved in the naming review / what do I need to do to kick that off?

I don't love the current package names, but they're the result of brainstorming with a lot of stakeholders and we didn't come up with anything that we thought was unambiguously better. 

I've read through REP-144, and as you say, I think we're very borderline.

**Comment by nuclearsandwich** [comment]
> @nuclearsandwich -- what's involved in the naming review / what do I need to do to kick that off?

Naming review happens as part of a package's first inclusion in the rosdistro database (this repository) so this release has started the process. As far as what's involved, those of us who are handling incoming rosdistro updates will review the names of new packages to make sure they adhere to REP-144 as part of this review process. Often packages pass without remark and it only comes up when something either violates the conventions or is right on the fence. In this case, I've sought a second opinion and if we're unable to find a consensus we'll pull in more members of the rosdistro maintenance team to confer.

> I don't love the current package names, but they're the result of brainstorming with a lot of stakeholders and we didn't come up with anything that we thought was unambiguously better.
> 
> I've read through REP-144, and as you say, I think we're very borderline.

The main reason I flagged the names is that without the context of `hydrographic`, which is part of the repository names but not the package names, I think they are too generic. One alternative would be to name the packages `hydrographic_acoustic_msgs` and `hydrographic_environmental_msgs` which would improve their specificity at the cost of extra verbosity.

**Comment by clalancette** [comment]
I agree that this is borderline.  I can buy the package name `acoustic_msgs`; that is pretty descriptive of what it does.  (though I might suggest `acoustic_sensor_msgs`, which is a bit more accurate)

But I don't think we should use `environmental_msgs` here.  Just think about how that package will  turn out in the distribution; it will be `ros-noetic-environmental-msgs`.  Which environment?  Underwater, in the air, in space, on the ground?  I think this is too generic a name, so we should change it to be more clear that it is for maritime uses, somehow or another.

**Comment by vatanaksoytezer** [comment]
Thank you for writing your input @clalancette! @lauralindzey what do you think? Do you think we can discuss this in a future maritime meeting or can you discuss with your colleagues?

**Comment by lauralindzey** [comment]
Thanks for the review @clalancette!

I've pinged Arjo about getting this on the agenda for this month's working group meeting, which will be held on 19 Jan.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/january-maritime-wg-meeting/29209/1

**Comment by lauralindzey** [comment]
I've discussed this with colleagues and a this month's maritime working group meeting.

We've agreed that we want to run with a prefix of "marine_*" for both message packages, with a view towards a future where they may be collected in a meta package `marine_msgs`.

We are also considering changing `environmental_msgs` to the more generic `marine_sensor_msgs`, given feedback from @rolker who is interested in contributing messages from a marine radar, in addition to the various scientific sensor messages that were originally envisioned. @nuclearsandwich @clalancette -- is the more generic name OK, or do y'all encourage more specificity when packaging messages? (The alternative would be `marine_environmental_msgs` and `marine_radar_msgs`.)

Once I get the OK that our proposed package names are acceptable, I'll update the repository with the new names.

**Comment by CaptKrasno** [comment]
Maybe:

- marine_sensor_msgs/maritime_sensor_msgs  corresponds to sensors related to normal ship operation and navigation(radars, AIS, NMEA 2000, etc.)  
- oceanographic_sensor_message corresponds to true oceanographic sensors (CTD, O2, Chlorophyll, ORP, etc.)

Where does this discussion leave us on the Acoustic_msgs?   Are we still ok with that?   

Where do future messages for Hydrographic Nav Sensors like heave sensors, tide gauges, INS go?    What about sensors that are acoustic and nav sensors like USBL and LBL.   How about sensors that are oceanographic, navigation, and hydrographic like a DVL/USBL?

I think the distinction for marine_sensor_msgs is fairly safe.   However,  the line blurs between oceanographic, environmental, acoustic and hydrographic.   I think we should consider if it is worth creating a distinction at all?   Maybe they all fall under the purview of oceanographic_msgs and we should only have one package?  I'm starting to feel like a simpler package structure would nullify discipline overlap and help reduce future code refactors.   

**I think we need to step back and ask ourselves what we are trying to distinguish with package names and why.**

**Comment by rolker** [comment]
> Where does this discussion leave us on the Acoustic_msgs? Are we still ok with that?

I think acoustic_msgs/* would become marine_acoustic_msgs/*.

> 
> Where do future messages for Hydrographic Nav Sensors like heave sensors, tide gauges, INS go? What about sensors that are acoustic and nav sensors like USBL and LBL. How about sensors that are oceanographic, navigation, and hydrographic like a DVL/USBL?
> 
> I think the distinction for marine_sensor_msgs is fairly safe. However, the line blurs between oceanographic, environmental, acoustic and hydrographic. I think we should consider if it is worth creating a distinction at all? Maybe they all fall under the purview of oceanographic_msgs and we should only have one package? I'm starting to feel like a simpler package structure would nullify discipline overlap and help reduce future code refactors.
> 
> **I think we need to step back and ask ourselves what we are trying to distinguish with package names and why.**

I think simplifying with exceptions is the way to go. The package name should identify that messages within are marine related so we don't have to name types themselves MarineThis.msg or MarineThat.msg. We could use hydrographic or maritime, but I like marine because it's shorter.

Maybe the default package should simply by marine_msgs or marine_sensor_msgs (to reserve marine_msgs as a meta package).

Reasons for exceptions would be for a collection of message types that might need to be grouped together without adding unnecessary length to the message type names. For example, the  messages type names in acoustic_msgs  would become more cumbersom if we had to rename PingInfo.msg to AcousticPingInfo.msg to distinguish it from a hypotetical RadarPingInfo.msg.

Message types that are more self contained would live the the default marine package.

So for messages that are ready to go:

marine_sensor_msgs/TurbityNTU.msg          // from environmental_msgs/
marine_acoustic_msgs/*.msg                        // from acoustic_msgs/*.msg

For messages that are currently under development, I have:

marine_sensor_msgs/RadarSector.msg
marine_sensor_msgs/RadarEcho.msg

Potential future messages:

marine_sensor_msgs/CTD.msg                     
marine_sensor_msgs/O2.msg                     
marine_sensor_msgs/Chlorophyll.msg
etc...

AIS messages are some I plan to work on soon. I'm not sure if they'll go into a separate package or not.

marine_sensor_msgs/AISPositionReport.msg
marine_sensor_msgs/AISBaseStationReport.msg
marine_sensor_msgs/AISShipData.msg
etc...

or 

marine_ais_msgs/PositionReport.msg
marine_ais_msgs/BaseStationReport.msg
marine_ais_msgs/ShipData.msg
etc...

**Comment by lindzey** [comment]
@CaptKrasno -- I like @rolker's reasoning here. (I also kind of see it as a question of namespacing -- once you have enough related messages + ecosystem around them, maybe it should be its own package rather than being thrown into an onmibus. For what we've done so far, I think the acoustic_msgs definitely fit in that category, while all of the simple one-value oceanographic sensors do not.)

So, I'd propose going with:
* marine_sensor_msgs for simple sensors (rename of environmental_msgs)
* marine_acoustic_msgs for acoustic_msgs  (rename of acoustic_msgs)
* TBD for the AIS and radar work Roland is doing -- I'd like more information / a better sense of how much of a message hierarchy and processing ecosystem is likely to build up around each.

I'm also happy to set up another real-time meeting with y'all if you think further discussion is warranted -- just let me know. I'm eager to keep this process moving along =)

**Comment by CaptKrasno** [comment]
@rolker @lauralindzey 

I largely like Roland's approach.    I think it's worth a quick chat though just to sync our thoughts (on this and in general) if you guys are up for it.

-Kris

**Comment by nuclearsandwich** [comment]
> @nuclearsandwich @clalancette -- is the more generic name OK, or do y'all encourage more specificity when packaging messages? (The alternative would be `marine_environmental_msgs` and `marine_radar_msgs`.)

There is still ongoing discussion and so I would not consider this a definitive review of any name but I think that `marine_sensor_msgs` is very much an improvement over `environmental_msgs`.

---

Regarding the discussion around simpler package names versus more precise package names: Packages are both the unit of installation/distribution in ROS and the unit of maintenance/administration. So creating packages that are about the size of your administrative unit (team or group for example) is the most straightforward thing to do, rather than trying to create one all-encompassing package that will have varied _owners_. Of course, having varied contributors is welcome and encouraged. Some care should also be taken that packages are small enough that most users will want nearly everything in the package or none of it so that they are not installing more than they need.

**Comment by lauralindzey** [comment]
@rolker, @CaptKrasno and I (the main contributors behind these packages) got together today to talk about the need to rename these packages. Our discussion converged, and was in line with what was discussed in the monthly maritime working group, so at this point I think we're ready for naming re-review. @nuclearsandwich @clalancette  -- what are our next steps?

Our proposal:

* rename repo itself hydrographic_msgs -> marine_msgs
* rename acoustic_msgs -> marine_acoustic_msgs
* rename environmental_msgs -> marine_sensor_msgs

A big point of discussion was how to decide which (soon-to-be-added) marine messages belong in which package. The guidelines we came up with are:
* The default category is marine_sensor_msgs
* Create a new package for:
   * anything complicated enough to require custom submessages (e.g. existing imaging&profiling sonar messages in marine_acoustic_msgs)
   * anything with a large number of tightly-related messages. (e.g. when developed, AIS will be its own package)

**Comment by clalancette** [comment]
> Our proposal:
> 
>     * rename repo itself hydrographic_msgs -> marine_msgs
> 
>     * rename acoustic_msgs -> marine_acoustic_msgs
> 
>     * rename environmental_msgs -> marine_sensor_msgs

That sounds very reasonable to me.  Your package names (once you release them) will end up being:

* `ros-noetic-marine-acoustic-msgs`
* `ros-noetic-marine-sensor-msgs`

And as far as I'm concerned, that well describes what they are for and should be REP-144 compliant.

**Comment by vatanaksoytezer** [comment]
@clalancette thank you very much for the quick review! Are we good to go ahead and do the renaming and re-releasing?

**Comment by nuclearsandwich** [comment]
> Are we good to go ahead and do the renaming and re-releasing?

Yes! Please close this PR when you do the new release and link this discussion in a comment on that PR so that the review team has the full context for the new release.

Thanks for iterating on this with us!

**Comment by mjcarroll** [comment]
Friendly ping @vatanaksoytezer

**Comment by lauralindzey** [comment]
I’m the holdup… on vacation now, this is on my list for early March when I
get back.

**Comment by vatanaksoytezer** [comment]
@lauralindzey let's sync when you are back and I am sure we can get this done in no time!

**Comment by clalancette** [comment]
Given that this needs a new PR to move forward, I'm going to close this to get it off of the rosdistro docket.  Do feel free to open a new one when things have been renamed.  Thanks for the understanding and iteration here.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/new-ros-enhancement-proposal-for-marine-robotics/36218/6

**Pull request closed without merging**
